src/lib/db/client-applications-store.ts

```diff
@@ -152,10 +152,28 @@ export default class ClientApplicationsStore
 
     async bulkUpsert(apps: Partial<IClientApplication>[]): Promise<void> {
         const rows = apps.map(remapRow);
+        const uniqueRows = Object.values(
```

Comment:
onConflict only respect conflicts with existing records but if the new input data has duplicates the insert would fail. By putting dedupe in the store we guarantee it is always correct